controllers/articles.controller.js

```diff
@@ -12,4 +12,19 @@ exports.getArticles = (req, res, next) => {
     selectArticles().then((articles) => {
         res.status(200).send({ articles })
     })
+}
+
+exports.getComments = (req, res, next) => {
+    const { article_id } = req.params
+    const commentPromises = [selectComments(article_id)]
+    if(article_id){
+        commentPromises.push(selectArticleByID(article_id))
+    }
```

Comment:
Is this if block necessary? I'd say it can go straight in your `commentPromises` array.
If there is no article_id, the error handling in `selectArticleById` will catch it, and your `Promise.all` will reject.